src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -70,11 +71,27 @@ internal static bool IsDirectoryTooLong(string fullPath)
         }
 
         /// <summary>
-        /// Adds the extended path prefix (\\?\) if not already present.
+        /// Adds the extended path prefix (\\?\) if not already present, IF the path is not relative,
+        /// AND the path is more than 259 characters. (> MAX_PATH + null)
+        /// </summary>
+        internal static string AddExtendedPathPrefixForLongPaths(string path)
```

Comment:
Wording with these is a bit awkward without getting really long. Technically it's AddExtendedPathPrefixForLongPathsIfNecessary (as it won't add it if it is already there). I'm hesitant to go that far. :)